Web.Framework/EndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Web.Framework
+{
+    public static class EndpointRouteBuilderExtensions
+    {
+        public static void MapHttpHandler<THttpHandler>(this IEndpointRouteBuilder builder, Action<HttpModel> configure = null)
+        {
+            var endpoints = HttpHandler.Build<THttpHandler>(configure);
```

Comment:
This is the most unrealistic thing in this PR. A real framework would have options n'such' that need to plug in here.